scalding-hraven/src/main/scala/com/twitter/scalding/hraven/reducer_estimation/HRavenHistoryService.scala

```diff
@@ -75,18 +75,29 @@ object HRavenHistoryService extends HistoryService {
    * TODO: query hRaven for successful jobs (first need to add ability to filter
    *       results in hRaven REST API)
    */
-  private def fetchSuccessfulFlows(client: HRavenRestClient, cluster: String, user: String, batch: String, signature: String, max: Int, nFetch: Int): Try[Seq[HRavenFlow]] = {
-    Try(client.fetchFlowsWithConfig(cluster, user, batch, signature, nFetch, RequiredJobConfigs: _*)).map { flows =>
-      val successfulFlows = flows.asScala.filter(_.getHdfsBytesRead > 0).take(max)
-      if (successfulFlows.isEmpty) {
-        LOG.warn("Unable to find any successful flows in the last " + nFetch + " jobs.")
+  private def fetchSuccessfulFlows(client: HRavenRestClient, cluster: String, user: String, batch: String, signature: String, max: Int, nFetch: Int): Try[Seq[Flow]] = {
+    Try(client.fetchFlowsWithConfig(cluster, user, batch, signature, nFetch, RequiredJobConfigs: _*))
+      .flatMap { flows =>
+        Try {
+          // Ugly mutable code to add task info to flows
+          flows.asScala.foreach { flow =>
+            flow.getJobs.asScala.foreach { job =>
+              val tasks = client.fetchTaskDetails(flow.getCluster, job.getJobId)
+              job.addTasks(tasks)
+            }
+          }
+
+          val successfulFlows = flows.asScala.filter(_.getHdfsBytesRead > 0).take(max)
+          if (successfulFlows.isEmpty) {
+            LOG.warn("Unable to find any successful flows in the last " + nFetch + " jobs.")
+          }
+          successfulFlows
+        }
+      } recoverWith {
```

Comment:
Also always use .'s between methods. (I see the old didn't have it, but it should have had)